src/mesh.cpp

```diff
@@ -116,19 +108,20 @@ void readMainFile(Mesh &mesh, const std::string &filename, const std::string &da
   }
 
   for (int i = 0; i < reader->GetUnstructuredGridOutput()->GetNumberOfCells(); i++) {
-    int cellType = reader->GetUnstructuredGridOutput()->GetCell(1)->GetCellType();
+    int cellType = reader->GetUnstructuredGridOutput()->GetCell(i)->GetCellType();
 
+    //Here we use static cast since VTK library returns a long long unsigned int however preCICE uses int for PointId's
     if (cellType == VTK_TRIANGLE) {
       vtkCell *             cell = reader->GetUnstructuredGridOutput()->GetCell(i);
-      std::array<size_t, 3> elem{cell->GetPointId(0), cell->GetPointId(1), cell->GetPointId(2)};
+      std::array<size_t, 3> elem{static_cast<int>(cell->GetPointId(0)), static_cast<int>(cell->GetPointId(1)), static_cast<int>(cell->GetPointId(2))};
```

Comment:
```suggestion
      std::array<int, 3> elem{static_cast<int>(cell->GetPointId(0)), static_cast<int>(cell->GetPointId(1)), static_cast<int>(cell->GetPointId(2))};
```
The array type needs to be int as well.

Reply:
> Can we actually assume that the point ids are positive and continuously numbered from 0?

The vertex IDs? No, there is no guarantee.

Reply:
I am not sure if they are continuous but they are `unsigned long int` type, so all positive.